### PR TITLE
Fix the coverage test

### PR DESCRIPTION
**Pull request by YiMysty**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java [hunk 1]
```diff
@@ -294,9 +296,77 @@ public ServiceCall answer(InvocationOnMock invocation) {
...
+                anyMapOf(String.class, String.class),
+                any(HttpClient.CallTemplate.class),
+                tokenExchangeServiceCallbackArgumentCaptor.capture());
+        TokenExchange.TokenExchangeServiceCallback tokenExchangeServiceCallback = tokenExchangeServiceCallbackArgumentCaptor.getValue();
+        assertNotNull(tokenExchangeServiceCallback);
+        String exceptionMessage = "Call to token exchange failed for whatever reason";
+        tokenExchangeServiceCallback.onCallFailed(new Exception(exceptionMessage));
+
+        /*
+         *  No retries and Cosmos DB does not get called
```
Comment (guperrot) [inline sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: Use dots at end of sentences.

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java [hunk 2]
```diff
@@ -501,6 +571,36 @@ public void createEndToEnd() {
...
+                any(HttpClient.CallTemplate.class),
+                tokenExchangeServiceCallbackArgumentCaptor.capture());
+        TokenExchange.TokenExchangeServiceCallback tokenExchangeServiceCallback = tokenExchangeServiceCallbackArgumentCaptor.getValue();
+        assertNotNull(tokenExchangeServiceCallback);
+        String exceptionMessage = "Call to token exchange failed for whatever reason";
+        tokenExchangeServiceCallback.onCallFailed(new Exception(exceptionMessage));
+        tokenExchangeServiceCallback.onCallFailed(new Exception(exceptionMessage));
+
+        /*
+         *  No retries and Cosmos DB does not get called
```
Comment (guperrot) [inline sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: dot missing

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java [hunk 3]
```diff
@@ -532,6 +632,36 @@ public void deleteEndToEnd() {
         // TODO: assert error is null and document is null
```
Comment (guperrot) [inline sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: how about this todo?

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java [hunk 4]
```diff
@@ -532,6 +632,36 @@ public void deleteEndToEnd() {
...
+                anyMapOf(String.class, String.class),
+                any(HttpClient.CallTemplate.class),
+                tokenExchangeServiceCallbackArgumentCaptor.capture());
+        TokenExchange.TokenExchangeServiceCallback tokenExchangeServiceCallback = tokenExchangeServiceCallbackArgumentCaptor.getValue();
+        assertNotNull(tokenExchangeServiceCallback);
+        String exceptionMessage = "Call to token exchange failed for whatever reason";
+        tokenExchangeServiceCallback.onCallFailed(new Exception(exceptionMessage));
+
+        /*
+         *  No retries and Cosmos DB does not get called
```
Comment (guperrot) [inline sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: dot missing

**Comment by YiMysty** [comment]
This pr fixed most of coverage test... but still some places is missing... Ask for advice here :)

**Comment by YiMysty** [comment]
not quite understand what this is for... :thinking: 
image

**Comment by YiMysty** [comment]
image
hmmm is there a way to stub the logging?

**Comment by YiMysty** [comment]
image
this method is literally not get called... but the function requires return :thinking:

**Comment by guperrot** [comment]
> not quite understand what this is for... 🤔
> image

You need a test that cancel operations with setEnabled(false) before they are completed.
From caller perspective:

future = storage.callSomething()
storage.setEnabled(false);

future.get() => does not hang
assert(future.get().getDocumentError() ....)

**Comment by guperrot** [comment]
> image
> hmmm is there a way to stub the logging?

You need to setup a mock that will trigger a http error.
The future not being completed on this method is puzzling me, I think your calls will hang on http error right? You need to handle futures appropriately in the code, @nikitag-ms has context on this and you should pair with.

**Comment by guperrot** [comment]
> image
> this method is literally not get called... but the function requires return 🤔

Why are you overriding service call in the first place? I don't understand what this method is trying to do.

**Comment by YiMysty** [comment]
> > not quite understand what this is for... thinking
> > image
> 
> You need a test that cancel operations with setEnabled(false) before they are completed.
> From caller perspective:
> 
> future = storage.callSomething()
> storage.setEnabled(false);
> 
> future.get() => does not hang
> assert(future.get().getDocumentError() ....)

seems hung.... :fox_face:

**Comment by YiMysty** [comment]
thanks~

**Pull request merged**